app/src/main/java/net/aiscope/gdd_app/ui/capture/CaptureImagePresenter.kt

```diff
@@ -6,10 +6,16 @@ import timber.log.Timber
 
 class CaptureImagePresenter(
     val view: CaptureImageView,
-    val repository: SampleRepository
+    val repository: SampleRepository,
+    var processingImageCapture: Boolean = false
 ) {
 
     fun handleCaptureImageButton(imageName: String) {
+        if (processingImageCapture){
+            Timber.tag("Taking Photo").d("Already processing photo")
+            return
+        }
+        processingImageCapture = true
         Timber.tag("Taking Photo").d("button pressed")
```

Comment:
Hmm, should we move this log above the `if`, for debugging crashes?

Reply:
Sure